kukur/source/sql.py

```diff
@@ -220,56 +215,110 @@ def get_data(
             )
         return pa.Table.from_pydict({"ts": timestamps, "value": values})
 
+    def __prepare_data_query(
+        self, selector: SeriesSelector, start_date: datetime, end_date: datetime
+    ) -> Tuple[str, List]:
+        assert self._config.data_query is not None
+
+        try:
+            query = self._config.data_query % selector.field
+        except TypeError:
+            query = self._config.data_query
+        params = [selector.tags[tag_name] for tag_name in self._config.tag_columns]
+        params.extend(
+            [
+                self.__format_date(start_date),
+                self.__format_date(end_date),
+            ]
+        )
+        if self._config.query_string_parameters:
+            query = query.format(*params)
+            params = []
+        return query, params
+
     def __search_names(
         self, selector: SeriesSearch
     ) -> Generator[SeriesSelector, None, None]:
         connection = self.connect()
         cursor = connection.cursor()
         cursor.execute(self._config.list_query)
 
-        for (series_name,) in cursor:
-            yield SeriesSelector(selector.source, series_name)
+        for tag_values in cursor:
+            tags = {
+                tag_name: tag_value
+                for tag_name, tag_value in zip(self._config.tag_columns, tag_values)
```

Comment:
we should validate the size of the two lists we `zip` here. I just accidentaly had more `tag_columns` than `tag_values` here and it did not complain.